drivers/sqlite/sqlite.go

```diff
@@ -0,0 +1,321 @@
+	config *Config
+}
+
+func WithInstance(dbInstance *sql.DB, config *Config) (drivers.Driver, error) {
+	driverConfig := mergeConfigs(config, getDefaultConfig())
+
+	conn, err := dbInstance.Conn(context.Background())
+	if err != nil {
+		return nil, &drivers.DatabaseError{Driver: driverName, Command: "grabbing_connection", OrigErr: err, Message: "failed to grab connection to the database"}
+	}
+
+	return &sqlite{config: driverConfig, conn: conn, db: dbInstance}, nil
+}
+
+func Open(filePath string) (drivers.Driver, error) {
+	customParams, err := drivers.ExtractCustomParams(filePath, configParams)
+	if err != nil {
+		return nil, &drivers.AppError{Driver: driverName, OrigErr: err, Message: "failed to parse custom parameters from url"}
+	}
+
+	sanitizedConnURL, err := drivers.RemoveParamsFromURL(filePath, configParams)
+	if err != nil {
+		return nil, &drivers.AppError{Driver: driverName, OrigErr: err, Message: "failed to sanitize url from custom parameters"}
+	}
+
+	sanitizedConnURL = strings.TrimSuffix(sanitizedConnURL, "?")
+
+	driverConfig, err := mergeConfigWithParams(customParams, getDefaultConfig())
+	if err != nil {
+		return nil, &drivers.AppError{Driver: driverName, OrigErr: err, Message: "failed to merge custom params to driver config"}
+	}
+
+	if _, err := os.Stat(sanitizedConnURL); errors.Is(err, os.ErrNotExist) {
+		return nil, &drivers.AppError{Driver: driverName, OrigErr: err, Message: "failed to open db file"}
+	}
+
+	db, err := sql.Open(driverName, sanitizedConnURL)
+	if err != nil {
+		return nil, &drivers.DatabaseError{Driver: driverName, Command: "opening_connection", OrigErr: err, Message: "failed to open connection with the database"}
+	}
+
+	conn, err := db.Conn(context.Background())
+	if err != nil {
+		return nil, &drivers.DatabaseError{Driver: driverName, Command: "grabbing_connection", OrigErr: err, Message: "failed to grab connection to the database"}
+	}
+
+	driverConfig.closeDBonClose = true
+
+	return &sqlite{
+		conn:   conn,
+		db:     db,
+		config: driverConfig,
+	}, nil
+}
+
+func (driver *sqlite) Ping() error {
+	ctx, cancel := drivers.GetContext(driver.config.StatementTimeoutInSecs)
+	defer cancel()
+
+	return driver.conn.PingContext(ctx)
+}
+
+func (sqlite) DriverName() string {
+	return driverName
+}
+
+func (driver *sqlite) Close() error {
+	if driver.conn != nil {
+		if err := driver.conn.Close(); err != nil {
+			return &drivers.DatabaseError{
+				OrigErr: err,
+				Driver:  driverName,
+				Message: "failed to close database connection",
+				Command: "sqlite_conn_close",
+				Query:   nil,
+			}
+		}
+	}
+
+	if driver.db != nil && driver.config.closeDBonClose {
+		if err := driver.db.Close(); err != nil {
+			return &drivers.DatabaseError{
+				OrigErr: err,
+				Driver:  driverName,
+				Message: "failed to close database",
+				Command: "sqlite_db_close",
+				Query:   nil,
+			}
+		}
+		driver.db = nil
+	}
+
+	driver.conn = nil
+	return nil
+}
+
+func (driver *sqlite) Lock() error {
```

Comment:
What do you think about using an atomic property to lock the driver? This would errors due concurrent calls from the outside (say calling the `Apply` method twice)